### ISSUE TITLE
How to restore the saver?

### ISSUE THREAD

**Issue by duanLH** [ppwwyyxx/tensorpack#171]
how to do when i want to restore the saver ？ thanks

**Comment by duanLH** [comment]
I think its better to add a choice to logger.auto_set_dir() to restore the last checkpoint

**Comment by ppwwyyxx** [comment]
Most examples have a `--load` option which can restore from a saver.
Basically set the `session_init` option in your training config, to `SaverRestore(...)`

**Comment by duanLH** [comment]
thanks, when i use --load, the parameters are restored by a checkpoint at 70 epoch, but  the start epoch is reset to 1.

**Comment by ppwwyyxx** [comment]
Yes. If you want to start from a different epoch number, you can use the [starting_epoch](http://tensorpack.readthedocs.io/en/latest/modules/train.html#tensorpack.train.TrainConfig) option.

**Comment by duanLH** [comment]
Yes, I means that when I restore a model in epoch N, the program is start with N, but the print is start epoch 1, I think the print should be change to N. (sorry for my poor english)

**Comment by ppwwyyxx** [comment]
If you use that option, the logs it will print will be changed to N.

**Comment by ppwwyyxx** [comment]
And what epoch number it prints usually doesn't have any effect to the training. The only difference it may make is when you use a epoch-number-based learning rate setter.

**Comment by duanLH** [comment]
OK .  Tensorflow is a flexible tool,  but there are still not have a unified coding standard。 thanks for your contributions very well,

**Comment by ppwwyyxx** [comment]
Yes. An "epoch" can have different semantics to different people.

But anyway, I think it's good if I could add a feature to (optionally) automatically recover the epoch number from log. I'll see what can be done.

**Comment by PatWie** [comment]
Wasn't there already an older version of tensorpack that saved the epoch number an global step into the checkpoint file?

**Comment by ppwwyyxx** [comment]
Was it? I don't remember I've saved the epoch number in checkpoint before.
